ChainSafe/lodestar
# Implement aggregated attestation pool

**Motivation**

+ We implemented aggregated attestation pool as a repository and it includes so many redundant attestations in lodestar proposed block
+ This issue was found in altair-devnet-1

**Description**

Implement aggregated attestation pool referring to other client's implementation:
+ Partition aggregated attestations by slot and attestation data
+ Preaggregate attestation if there is no intersection regarding attestingIndices (or aggregationBits) to maximize profit
+ Use `attestingIndices` instead of `aggregationBits` for better performance
+ Remove attestations from seen attestations that are included in block chain
+ Validate all attestations before we include in block
+ Sort attestations by inclusion distance, then by number of attesters
+ Prune per slot
+ api should use it instead of the unaggregated attestation pool
+ Remove `aggregationAndProof` repository

Closes #2845

**Test**
Tested with prater by the api
